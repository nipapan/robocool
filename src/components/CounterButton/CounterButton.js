import React from 'react';

class CounterButton extends React.Component {
   constructor() {
      super();
      this.state = { counter: 1 }
   }

   render() {
      return (
         <button
            id='counter'
            className='f6 link dim br1 ph3 pv2 mb2 dib white bg-dark-gray'
            onClick={() => this.setState(state => ({counter: state.counter+1}) )}>
            Counter: {this.state.counter}
         </button>
      );
   }
}

export default CounterButton;
