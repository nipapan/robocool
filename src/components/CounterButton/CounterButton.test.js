import {shallow} from 'enzyme';
import React from 'react';
import CounterButton from "./CounterButton";

it('renders Counter Button', () => {
   expect(shallow(<CounterButton />)).toMatchSnapshot();
})

it('increases Counter correctly', () => {
   const wrapper = shallow(<CounterButton />);

   wrapper.find('[id="counter"]').simulate('click');
   wrapper.find('[id="counter"]').simulate('click');
   expect(wrapper.state()).toEqual({counter:3});
})