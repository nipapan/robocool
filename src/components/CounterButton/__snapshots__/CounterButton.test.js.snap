// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Counter Button 1`] = `ShallowWrapper {}`;
